packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,899 @@
+/// provided to the [RegularWindow] widget, who does the work of rendering the
+/// content inside of this window.
+///
+/// The user of this class is responsible for managing the lifecycle of the window.
+/// When the window is no longer needed, the user should call [destroy] on this
+/// controller to release the resources associated with the window.
+///
+/// An example usage might look like:
+///
+/// ```dart
+/// final RegularWindowController controller = RegularWindowController(
+///   preferredSize: const Size(800, 600),
+///   preferredConstraints: const BoxConstraints(minWidth: 640, minHeight: 480),
+///   title: 'Example Window',
+/// );
+/// runWidget(
+///   RegularWindow(
+///     controller: controller,
+///     child: MaterialApp(home: Container()),
+///   ),
+/// );
+/// ```
+///
+/// Children of a [RegularWindow] widget can access the [RegularWindowController]
+/// via the [WindowControllerScope] inherited widget.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+@internal
+abstract class RegularWindowController extends BaseWindowController {
+  /// Creates a [RegularWindowController] with the provided properties.
+  ///
+  /// Upon construction, the window is created by the platform.
+  ///
+  /// The [preferredSize] is the preferred content size of the window.
+  /// This might not be honored by the platform. This is the size that
+  /// the platform will try to apply to the window when it is created. In contrast,
+  /// the [preferredConstraints] field enforces the minimum and maximum size of
+  /// the window. If the [preferredSize] does not satisfy the [preferredConstraints]
+  /// or the [preferredSize] is null, then the platform will attempt to use an
+  /// initial size that does satisfy the [preferredConstraints] instead.
+  ///
+  /// The [preferredConstraints] are the constraints placed upon the size
+  /// of the window. This might not be honored by the platform.
+  /// This field enforces a minimum and maximum size on the window. If the
+  /// user attempts to resize the window beyond these constraints, the platform
+  /// will enforce the constraints according to its own policy. For example, the
+  /// platform might clip the content to fit within the resized window, or it might
+  /// prevent the window from being resized altogether. If null, the window will
+  /// be unconstrained.
+  ///
+  /// If both [preferredSize] and [preferredConstraints] are null,
+  /// then the platform will use its own default size for the window.
+  ///
+  /// The [title] argument configures the window's initial title.
+  /// If omitted, some platforms might fall back to the app's name.
+  ///
+  /// The [delegate] argument can be used to listen to the window's
+  /// lifecycle. For example, it can be used to save state before
+  /// a window is closed.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  factory RegularWindowController({
+    Size? preferredSize,
```

Comment:
So this will create regular window that is sized to contents by default? Unconstrained content usually needs special care in Flutter, I can see this behavior being confused when users immediately get layout exceptions.